### ISSUE TITLE
Generated builder won't accept NULL as parameter for Optional<> field

### ISSUE THREAD

**Issue by skordes** [immutables/immutables#781]
Using version 2.6.1

When defining a property of type java.util.Optional<> (Java 8), the generated Immutable class has a method to set a value for that property. However, this method won't accept NULL values.

For example, I have a class that has only one Optional field. The class is annotated with `@Value.Immutable`.

```java
package org.immutables.bug;

import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
public interface System {

    Optional<String> getReference();

}
```

When setting a `NULL` value, an `NullPointerException` is thrown:

```java
package org.immutables.bug;

import org.junit.Assert;
import org.junit.Test;

import java.util.Optional;

public class SystemTest {

    @Test
    public void testBuilderWithNonnullValue() {
        final String ref1 = "ref";
        final System s1 = ImmutableSystem
                .builder()
                .reference(ref1)
                .build();
        Assert.assertTrue(s1.getReference().isPresent());
        Assert.assertEquals(ref1, s1.getReference().get());
    }

    @Test
    public void testBuilderWithNullValue() {
        final String ref2 = null;
        final System s2 = ImmutableSystem
                .builder()
                .reference(ref2)// error
                .build();
        Assert.assertFalse(s2.getReference().isPresent());
        Assert.assertEquals(Optional.empty(), s2.getReference());
    }

    @Test
    public void testBuilderWithEmptyOptional() {
        final System s2 = ImmutableSystem
                .builder()
                .reference(Optional.empty())// no error
                .build();
        Assert.assertFalse(s2.getReference().isPresent());
        Assert.assertEquals(Optional.empty(), s2.getReference());
    }

}
```

``` stacktrace
java.lang.NullPointerException: reference
	at java.util.Objects.requireNonNull(Objects.java:228)
	at org.immutables.bug.ImmutableSystem$Builder.reference(ImmutableSystem.java:158)
	at org.immutables.bug.SystemTest.testBuilderWithNullValue(SystemTest.java:21)
	...
```

This is because the generated code, still calls `java.util.Objects.requireNonNull(...)`, which is undesirable.

**Comment by elucash** [comment]
This is by design, you can read some older discussions about this (#245 #152)
tl;dr:
There's style `@Value.Style(optionalAcceptNullable=true)` for this.

In addition to that, the  "encodings" functionality (#363  http://immutables.github.io/encoding.html ) could possibly allow custom ways to code any special kind of attribute, including redefining how optionals are supported.
Thank you!

**Comment by flaffyyeti** [comment]
that cause most common bug in our that coder puts value in optional immutable fields witch could contain null.
It so easy to do and forgot to wrap in Nullable.
that have burned tons of money for business.
